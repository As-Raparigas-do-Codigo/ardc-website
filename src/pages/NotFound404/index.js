import React from "react";
import { Row, Container, Col } from "react-bootstrap";
import LinkButton from "components/LinkButton";
import PageLayout from "components/PageLayout";
import notFoundImg from "assets/erro404-image.svg"
import { SiteContent } from "Constants";

const NotFound404 = () => {
  return (
    <PageLayout>
      <Container className="info padding-top-bottom-large align-center">
        <Row>
          <Col>
            <img src={notFoundImg} alt="404 warning" />
          </Col>
          <Col>
            <Row>
              <p className="text-secondary-color text-700-weight">Erro 404</p>
            </Row>
            <Row>
              <h4>{SiteContent.Title.NotFound404}</h4>
            </Row>
            <Row>
              <p>{SiteContent.Subtitle.NotFound404}</p>
            </Row>
            <Row>
              <p>
                <LinkButton
                  href="/">
                  {SiteContent.Buttons.NotFound404}
                </LinkButton>
              </p>
            </Row>
          </Col>
        </Row>
      </Container>
    </PageLayout>
  );
};

export default NotFound404;
