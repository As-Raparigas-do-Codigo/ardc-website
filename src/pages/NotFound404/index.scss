.notfound404 {
  &__row > * {
    display: flex;
    justify-content: center;
  }

  &__image-wrapper {
    padding: 0 20px;

    img {
      max-width: 100%;
    }
  }

  &__text {
    flex-direction: column;
  }
}
