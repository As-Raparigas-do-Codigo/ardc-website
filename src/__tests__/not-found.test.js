import NotFound404 from 'pages/NotFound404';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

const translationProps = {
  translation: jest.fn()
};

it('renders correctly', () => {
  const tree = renderer.create(<NotFound404 {...translationProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
