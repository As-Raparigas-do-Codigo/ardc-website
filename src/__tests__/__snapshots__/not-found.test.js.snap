// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NotFound404.js renders correctly 1`] = `
<div>
  <div
    className="bg-blue"
  >
    <div
      className="pt-md-5 pb-4 container"
    />
  </div>
  <div>
    <div
      className="info padding-top-bottom-large align-center container"
    >
      <div
        className="notfound404__row row row-cols-lg-2 row-cols-md-1 row-cols-1"
      >
        <div
          className="notfound404__image-wrapper  col"
        >
          <img
            alt="404 warning"
            src="erro404-image.svg"
          />
        </div>
        <div
          className="notfound404__text col"
        >
          <div
            className="row"
          >
            <p
              className="text-secondary-color text-700-weight"
            >
              Erro 404
            </p>
          </div>
          <div
            className="row"
          >
            <h4
              className="notfound404__title"
            />
          </div>
          <div
            className="row"
          >
            <p />
          </div>
          <div
            className="row"
          >
            <p>
              <a
                className="button-primary"
                disabled={false}
                href="/"
                rel="noreferrer"
                target=""
              />
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing page rendering with translation should render page with translation 1`] = `
<h4
  class="notfound404__title"
/>
`;

exports[`testing page rendering without translation should render 404 page without translation 1`] = `
<h4
  class="notfound404__title"
/>
`;
